paytrail/views.py

```diff
@@ -292,9 +294,47 @@ def render_payment_details(self, request, **kwargs):
 
 
 def notification(request, token):
-    """Paytrail server calls this when payment was successful."""
+    """
+    Paytrail server calls this when payment was successful.
+    This can be called multiple times per one payment.
+    Creates a transaction object if not already created.
+    """
     validate_signature(request.GET)
-    get_model('order', 'Order').objects.get(
-        number=signer.unsign(token)
-    ).sources.get(source_type=get_source_type()).debit()
+    order = get_object_or_404(Order, number=signer.unsign(token))
+
+    transaction_id = request.GET.get('checkout-transaction-id', None)
+    if not transaction_id:
+        logger.error('checkout-transaction-id missing', code=400)
+        raise ValidationError
```

Comment:
Should there be an error message as in the next check?

Reply:
Ah, silly me. Fixed.